mypy.ini

```diff
@@ -4,49 +4,59 @@
 warn_unused_configs = True
 follow_imports = silent
 show_error_context = True
-files = **/numba/core/types/*.py, **/numba/core/datamodel/*.py, **/numba/core/rewrites/*.py, **/numba/core/unsafe/*.py
+show_error_codes = True
+files = **/numba/core/types/*.py, **/numba/core/typeconv/*.py, **/numba/core/datamodel/*.py, **/numba/core/rewrites/*.py, **/numba/core/unsafe/*.py
+ignore_errors = True
```

Comment:
Why is this needed? And, is it ignoring all errors?

Reply:
Previously we used the global level `files` setting to control which files to check with `mypy`.  This required specifying files in two places (here and below in the respective level).  This approach also doesn't seem to work with `pre-commit`.  My guess is that it's because `pre-commit` specifies the files to run on when invoking `mypy`, which [supercedes top-level config file options](https://mypy.readthedocs.io/en/stable/config_file.html#config-file-format).

This PR changes how we filter files.  Now, we always have `mypy` run on all of `numba/*`.  Levels 1-3 specify `ignore_errors = False`, so we do not ignore all errors.  For any file that does not fit one of the more specific patterns, we fall back to Level 4, which does have `ignore_errors = True`.  This way a file doesn't need to be specified twice in `mypy.ini` to get type checking, and it also works with `pre-commit` or when running `mypy` manually on specific files or directories.